### ISSUE TITLE
Opening a dialog with optional information, then acting on that information during a save/delete

### ISSUE THREAD

**Issue by lquantz** [volkanceylan/Serenity#1625]
Is there a good way to pass information from a dialog class to the corresponding repository classes, so we can take action on them if desired?

In my case I have a dialog that I would like to pass some optional information to under some circumstances, which ideally would cause an additional step to be taken in the repository, to add/update/delete a separate entity.

As an example, assume I'm working with automobiles and have a dialog class which I added a public boolean variable to, called "extras", and use it like this sometimes:

```
var dlg = new MyNamespace.AutomobileDialog();
dlg.extras = true;
```

I can then open the dialog with `dlg.loadEntityAndOpenDialog(<MyNamespace.AutomobileRow>{...})`.  

What I'd like to do is act on that 'extras' variable during a save/update/delete action, so that if the 'extras' variable is true in the `AutomobileDialog`, then in the `AutomobileRepository `class create an `AutomobileExtrasRow` entity and either update or save or delete it, depending on what's happening in the `AutomobileRow`.  To do this I need the dialog class to tell the repository class about the 'extras' variable in some way.

If there's a better way to do this kind of thing I'm open to that as well.

**Comment by volkanceylan** [comment]
You add a non-mapped field to your entity, populate that, or define a custom save request, deriving from default one. See existing ones already there.

**Comment by lquantz** [comment]
Thanks Volkan.  I assume for an example of a custom save request I should be looking in the NotesBehavior class, or did you have another one in mind in addition?

**Comment by volkanceylan** [comment]
It doesn't have to be behavior. There is a chapter in movie tutorial about custom list requests.

**Comment by lquantz** [comment]
I took your first suggestion and added a [NotMapped] field to my AutomobileRow class:

```
        [NotMapped]
        public Int32? Extras
        {
            get { return Fields.Extras[this];  }
            set {  Fields.Extras[this] = value; }
        }
```
In my UI code I open my dialog with (for example):
`dlg.loadEntityAndOpenDialog(<MyNamespace.AutomobileRow>{Extras: 35});`

...and the dialog opens.  I set a breakpoint in the dialog's `afterLoadEntity()` method and in there, `this.entity.Extras` is correctly set to the value I passed when opening the dialog.

When I hit the save button however and set a breakpoint in the AutomobileRepository's MySaveHandler class's `AfterSave() `method, `this.Row.Extras` was always null.   

That seemed odd, so I did some re-checking and remembered the dialog classes have a `getSaveEntity()` method, which I overrode and set the value of `Extras` in there.  After doing so the AutomobileRow class _did_ start getting the correct value of `this.Row.Extras` in its `AfterSave()` method.  

Not sure if my approach was the best way to go about it, but it seems to work?
